docs/usage/dns_names.md

```diff
@@ -25,95 +25,121 @@ spec:
+You should :
+- Have created a shoot cluster
+- Have created and correctly configured a DNS Provider (Please consult [this page](./dns_providers.md) for more information)
+- Have a basic understanding of DNS (see link under [References](#references))
 
-Gardener can only manage DNS records on your behalf if you have proper DNS providers in place. Please consult [this page](./dns_providers.md) for more information.
+There are 2 types of DNS that you can use within Kubernetes : 
+- internal (usually managed by coreDNS)
+- external (managed by a public DNS provider). 
 
-## Request DNS records via Service/Ingress resources
+This page, and the extension, exclusively works for external DNS handling.
 
-To request a DNS name for an Ingress or Service object in the shoot cluster
-it must be annotated with the DNS class `garden` and an annotation denoting
-the desired DNS names.
+Gardener allows 2 way of managing your external DNS:
+- Manually, which means you are in charge of creating / maintaining your Kubernetes related DNS entries
+- Via the Gardener DNS extension
+
+## Gardener DNS extension
+The managed external DNS records feature of the Gardener clusters makes all this easier. You do not need DNS service provider specific knowledge, and in fact you do not need to leave your cluster at all to achieve that. You simply annotate the Ingress / Service that needs its DNS records managed and it will be automatically created / managed by Gardener.
+
+Managed external DNS records are supported with the following DNS provider types:
+- aws-route53
+- azure-dns
+- azure-private-dns
+- google-clouddns
+- openstack-designate
+- alicloud-dns
+
+### Request DNS records for Ingress resources
+
+To request a DNS name for an Ingress or Service object in the shoot cluster it must be annotated with the DNS class `garden` and an annotation denoting the desired DNS names.
 
 Example for an annotated Ingress resource:
 
 ```yaml
-apiVersion: networking.k8s.io/v1
+apiVersion: networking.k8s.io/v1beta1
 kind: Ingress
 metadata:
+  name: amazing-ingress
   annotations:
-    dns.gardener.cloud/dnsnames: '*' # collects domains names from .spec.rules[].host
+    # Let Gardener manage external DNS records for this Ingress.
+    dns.gardener.cloud/dnsnames: special.example.com # Use "*" to collects domains names from .spec.rules[].host
+    dns.gardener.cloud/ttl: "600"
     dns.gardener.cloud/class: garden
-    # If you are delegating the certificate management to Gardener, uncomment the following line
-    #cert.gardener.cloud/purpose: managed
-  name: test-ingress
-  namespace: default
 spec:
   rules:
-    - host: test.ingress.my-dns-domain.com
-      http:
-        paths:
-          - backend:
-              service:
-                name: my-service
-                port:
-                  number: 9000
-            path: /
-            pathType: Prefix
-  tls:
-    - hosts:
-        - test.ingress.my-dns-domain.com
-      secretName: my-cert-secret-name
```

Comment:
would be nice to keep the tls section and the commented annotation `#cert.gardener.cloud/purpose: managed` to have a complete example as most users will use both shoot-dns-service and shoot-cert-service